libs/agno/agno/os/interfaces/a2a/utils.py

```diff
@@ -0,0 +1,445 @@
+
+    # 1. Handle output content
+    if run_output.content:
+        parts.append(Part(root=TextPart(text=run_output.content)))
+
+    # 2. Handle output media
+    artifacts: List[Artifact] = []
+    if run_output.images:
+        for idx, img in enumerate(run_output.images):
+            artifact_parts = []
+            if img.url:
+                artifact_parts.append(Part(root=FilePart(file=FileWithUri(uri=img.url, mime_type="image/jpeg"))))
+            artifacts.append(
+                Artifact(
+                    artifact_id=str(uuid4()),
+                    name=f"image_{idx}",
+                    description=f"Generated image {idx}",
+                    parts=artifact_parts,
+                )
+            )
+    if run_output.videos:
+        for idx, vid in enumerate(run_output.videos):
+            artifact_parts = []
+            if vid.url:
+                artifact_parts.append(Part(root=FilePart(file=FileWithUri(uri=vid.url, mime_type="video/mp4"))))
+            artifacts.append(
+                Artifact(
+                    artifact_id=str(uuid4()),
+                    name=f"video_{idx}",
+                    description=f"Generated video {idx}",
+                    parts=artifact_parts,
+                )
+            )
+    if run_output.audio:
+        for idx, aud in enumerate(run_output.audio):
+            artifact_parts = []
+            if aud.url:
+                artifact_parts.append(Part(root=FilePart(file=FileWithUri(uri=aud.url, mime_type="audio/mpeg"))))
+            artifacts.append(
+                Artifact(
+                    artifact_id=str(uuid4()),
+                    name=f"audio_{idx}",
+                    description=f"Generated audio {idx}",
+                    parts=artifact_parts,
+                )
+            )
+    if run_output.files:
+        for idx, file in enumerate(run_output.files):
+            artifact_parts = []
+            if file.url:
+                artifact_parts.append(
+                    Part(
+                        root=FilePart(
+                            file=FileWithUri(uri=file.url, mime_type=file.mime_type or "application/octet-stream")
+                        )
+                    )
+                )
+            artifacts.append(
+                Artifact(
+                    artifact_id=str(uuid4()),
+                    name=getattr(file, "name", f"file_{idx}"),
+                    description=f"Generated file {idx}",
+                    parts=artifact_parts,
+                )
+            )
+
+    # 3. Build the A2A message
+    agent_message = A2AMessage(
+        message_id=str(uuid4()),
+        role=Role.agent,
+        parts=parts,
+        context_id=run_output.session_id,
+        task_id=run_output.run_id,
+    )
+
+    # 4. Build and return the A2A task
+    return Task(
+        id=run_output.run_id or str(uuid4()),
+        context_id=run_output.session_id or str(uuid4()),
+        status=TaskStatus(state=TaskState.completed),
```

Comment:
We have other status right? So cancelled, error, etc?

Reply:
Added logic to handle critical errors and propagate them in the A2A conventional way. Also cancelled events. Good one!